builtin/read_osh.py

```diff
@@ -174,6 +175,16 @@ def _ReadPortion(delim_byte, max_chars, cmd_ev):
             eof = True
             break
 
+        elif backslash:
+            backslash = False
+            if ch == pyos.NEWLINE_CH:
+                continue
+            ch_array.append(pyos.BACKSLASH_CH)
+            ch_array.append(ch)
+        elif allow_escape and ch == pyos.BACKSLASH_CH:
+            backslash = True
```

Comment:
OK one thing I notice is that in _AppendParts(), we handle the `\` for the `read -d delim` case, with `span_e.Backslash`

Let me see if we can make the cases more consistent, i.e.  `(delim, -n) x (raw, not raw)`

Reply:
Oh now I see there is a `assert done`, so I think the `while True` loop can be removed?  And both return values `done, join_next` are unnecessary?

If so, this change seems like it can simplify the code, which is nice

Reply:
Yes, and the `while` loop is removed in the next PR #2353. I just tried to minimize the change.

Reply:
If you did it in the next change, it's OK, we can do it then

I made some other comments about the spec test assertions, which I think will not conflict with the next change

Thanks